File: hw/chalan/u2/hw19Inheritance/Farmsystem Python/model/Sheep.py
from datetime import datetime
from model.FarmAnimal import FarmAnimal

class Sheep(FarmAnimal):
    def __init__(self, last_shearing, id, breed, born_on, gender, is_able_to_reproduce, weight, cage):
        super().__init__(id, breed, born_on, gender, is_able_to_reproduce, weight, cage)
        self.last_shearing = last_shearing

    def shear(self):
        self.last_shearing = datetime.now()

    def __str__(self):
        return f"Sheep{{last_shearing={self.last_shearing}, parent={super().__str__()}}}"

File: hw/cevallos/u2/hw19FarmSystem/Java/FarmSystem/src/main/java/ec/edu/espe/farmsystem/model/Sheep.java
package ec.edu.espe.farmsystem.model;

import java.util.Date;

/**
 *
 * @author Mateo Cevallos
 */
public class Sheep extends FarmAnimal{
    private Date lastSheering;

    @Override
    public String toString() {
        return "Sheep{" + "lastSheering=" + getLastSheering() + "," + super.toString() + '}';
    }
    
    public Sheep(Date lastSheering, int id, String breed, Date bornOn, String gender, boolean isAbleToReproduce, float weight, Cage cage) {
        super(id, breed, bornOn, gender, isAbleToReproduce, weight, cage);
        this.lastSheering = lastSheering;
    }
    
    public void cutWhool(){
       
    }
    
    public void shear(){
        
    }

    /**
     * @return the lastSheering
     */
    public Date getLastSheering() {
        return lastSheering;
    }

    /**
     * @param lastSheering the lastSheering to set
     */
    public void setLastSheering(Date lastSheering) {
        this.lastSheering = lastSheering;
    }
    
}
